test/dnf/test_product_id.py

```diff
@@ -0,0 +1,97 @@
+
+    CACHE_FILE = "productid_repo_mapping.json"
+
+    def setUp(self) -> None:
+        self.temp_directory: Union[None, tempfile.TemporaryDirectory] = None
+
+    def tearDown(self) -> None:
+        if self.temp_directory is not None:
+            self.temp_directory.cleanup()
+
+    def _create_cache_file(self) -> str:
+        """
+        Create temporary directory and return path of cache file
+        """
+        self.temp_directory = tempfile.TemporaryDirectory()
+        return os.path.join(self.temp_directory.name, self.CACHE_FILE)
+
+    def _create_cache_file_read_only(self) -> str:
+        """
+        Create read-only temporary directory and return path of cache file
+        """
+        self.temp_directory = tempfile.TemporaryDirectory()
+        # Make directory read only
+        os.chmod(self.temp_directory.name, 0x400)
+        return os.path.join(self.temp_directory.name, self.CACHE_FILE)
+
+    def test_create_cache_file(self):
+        """
+        Test that it is possible to create cache file
+        """
+        cache_file = self._create_cache_file()
+        DnfProductManager._write_cache_file({"foo": "bar"}, cache_file)
+        data = DnfProductManager._read_cache_file(cache_file)
+        self.assertEqual(data, {"foo": "bar"})
+
+    def test_create_cache_file_in_read_only_directory(self):
+        """
+        Test that no exception is raised, when dnf plugin tries to create cache file
+        in directory that is read-only
+        """
+        cache_file = self._create_cache_file_read_only()
+        try:
+            DnfProductManager._write_cache_file({"foo": "bar"}, cache_file)
+        except PermissionError as err:
+            self.fail(f"DnfProductManager._write_cache_file(read_only_path) raised exception: {err}")
+
+    def test_read_cache_perm(self):
+        """
+        Test that no exception is raised, when dnf plugin tries to read cache file
+        that process cannot read due to restricted permissions
+        """
+        cache_file = self._create_cache_file()
+        DnfProductManager._write_cache_file({"foo": "bar"}, cache_file)
+        os.chmod(cache_file, 0x000)
+        try:
+            data = DnfProductManager._read_cache_file(cache_file)
+        except PermissionError as err:
+            self.fail(f"DnfProductManager._read_cache_file(read_only_path) raised exception: {err}")
+        else:
+            self.assertIsNone(data)
+
+    def test_read_non_existing_cache(self):
+        """
+        Test that no exception is raised, when dnf plugin tries to read non-existing cache file
+        """
+        cache_file = self._create_cache_file()
+        DnfProductManager._write_cache_file({"foo": "bar"}, cache_file)
+        os.chmod(cache_file, 0x000)
+        try:
+            data = DnfProductManager._read_cache_file(cache_file)
+        except PermissionError as err:
+            self.fail(f"DnfProductManager._read_cache_file(read_only_path) raised exception: {err}")
```

Comment:
is the exception checking needed here? if an exception is raisen and the test does not expect any, the test will properly fail